src/UI/Core/Window.h

```diff
@@ -19,7 +19,7 @@ class Window : public UIContainer
 	void onMouseDown(NAS2D::EventHandler::MouseButton button, int x, int y);
 	void onMouseUp(NAS2D::EventHandler::MouseButton button, int x, int y);
 	void onMouseMotion(int x, int y, int dX, int dY);
-
+	static inline constexpr float getWindowTitleBarHeight() { return sWindowTitleBarHeight; };
```

Comment:
I would probably have just moved the constant to the header file as a `protected` `static const` member. That would have been easier than introducing a new `get` method.

As it stands now, the constant was moved to the header file anyway, it just got put in the `private` section.

Reply:
I was unsure whether or not the derived classes were allowed to *change* the value since it was originally a local variable, so I went the safe route.

Reply:
I'm not sure I follow.

The original seems to be a `const` declared in the .cpp file. By moving it into the header as a protected member, it becomes accessible to subclasses. It's still just a `static const` though. Subclasses can choose if they want to use it or not, or define their own constant.

Reply:
It's safer to code with only as much access as needed. Only bumping it up on a case-by-case basis as needed.

In the future if a class `WindowBorderless` gets created or a derived window needs the option to create a window without a title bar, then `sWindowTitleBarHeight` can no longer be `const` nor `static` and the variable can move.

Right now, it acts as if it were a compile-time literal value, with all the compiler optimization that entails.